packages/@aws-cdk/aws-ec2/lib/machine-image.ts

```diff
@@ -269,8 +388,31 @@ export class AmazonLinuxImage extends GenericSSMParameterImage {
       storage,
     ].filter(x => x !== undefined); // Get rid of undefineds
 
-    const parameterName = '/aws/service/ami-amazon-linux-latest/' + parts.join('-');
-    super(parameterName, OperatingSystemType.LINUX, props.userData);
+    return '/aws/service/ami-amazon-linux-latest/' + parts.join('-');
+  }
+
+  private readonly cachedInContext: boolean;
+
+  constructor(private readonly props: AmazonLinuxImageProps = {}) {
+    super(AmazonLinuxImage.ssmParameterName(props), OperatingSystemType.LINUX, props.userData);
+
+    this.cachedInContext = props.cachedInContext ?? false;
+  }
+
+  /**
+   * Return the image to use in the given context
+   */
+  public getImage(scope: Construct): MachineImageConfig {
```

Comment:
why does this need to override the superclass implementation? I can't see any differences

Reply:
This one has the check on using the cache parameter or not.

Reply:
oh damn it was GenericSSMParameterImage, not GenericSsmParameterImage